src/main/java/org/opensrp/web/GzipBodyDecompressFilter.java

```diff
@@ -130,7 +124,23 @@ public GzippedInputStreamWrapper(final HttpServletRequest request) throws IOExce
 		public ServletInputStream getInputStream() throws IOException {
 			final ByteArrayInputStream sourceStream = new ByteArrayInputStream(bytes);
 			return new ServletInputStream() {
-				
+				private ReadListener readListener;
+
+				@Override
+				public boolean isFinished() {
+					return false;
+				}
+
+				@Override
+				public boolean isReady() {
+					return false;
+				}
+
+				@Override
+				public void setReadListener(ReadListener readListener) {
+					this.readListener = readListener;
```

Comment:
I also believe that this listener should be called somewhere in this implementation